core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/shared/services/dot-edit-page-resolver/dot-edit-page-resolver.service.ts

```diff
@@ -100,4 +89,16 @@ export class DotEditPageResolver implements Resolve<DotPageRenderState> {
             return of(dotRenderedPageState);
         }
     }
+
+    private getDotPageRenderOptions(route: ActivatedRouteSnapshot): DotPageRenderOptions {
+        return {
+            url: route.queryParams.url,
+            ...(route.queryParams.mode && { mode: route.queryParams.mode }),
+            ...(route.queryParams.language_id && {
+                viewAs: {
+                    language: route.queryParams.language_id
+                }
+            })
+        };
+    }
```

Comment:
This is hard to read, I suggest keep it simple:

```suggestion
private getDotPageRenderOptions(route: ActivatedRouteSnapshot): DotPageRenderOptions {
    const queryParams = route.queryParams;
    const options: DotPageRenderOptions = {
        url: queryParams.url,
    };
    if (queryParams.mode) {
        options.mode = queryParams.mode;
    }
    if (queryParams.language_id) {
        options.viewAs = {
            language: queryParams.language_id
        };
    }
    return options;
}
```